### PR TITLE
[CI] remove run_tests_torch_and_tf

### PR DESCRIPTION
**Pull request by sshleifer**
Problem:
This job is rerunning all tests, after `run_tests_torch` and `run_tests_tf` with both packages installed. Meaning that it ~doubles the runtime of CI and doesn't find very many bugs.

Solution: This changes `run_tests_torch_and_tf` to only run the tests that require both TF and torch

### CODE REVIEW

**Comment by sshleifer** [comment]
(fixed history locally for future PRs)

**Review by thomwolf — CHANGES_REQUESTED** [review]
We have several methods that require both torch and tf to be present at the same time (conversion methods in particular).

The goal of this test setup is to test them.

File: .circleci/config.yml [hunk 1]
```diff
@@ -12,7 +12,8 @@ jobs:
             - checkout
             - run: sudo pip install .[sklearn,tf-cpu,torch,testing]
             - run: sudo pip install codecov pytest-cov
-            - run: python -m pytest -n 8 --dist=loadfile -s -v ./tests/ --cov
+            - run: python -m pytest -n 8 --dist=loadfile -s -v ./tests/test_modeling_tf* -k model_equivalence --cov
+            - run: python -m pytest -sv ./tests/ -k batch_encode_plus_tensors --cov
```
Comment (sshleifer) [inline .circleci/config.yml:16]: this is faster without parralellism

**Comment by sshleifer** [comment]
I found two common tests that require both packages:
```
test_batch_encode_plus_tensors
test_pt_tf_model_equivalence
```
So I run test_pt_tf_model_equivalence for all test_modeling_tf* files, and
test_batch_encode_plus_tensors for all files. 
Still saves quite a bit of time.

**Comment by julien-c** [comment]
Kinda reluctant to do this change because then we need to manually maintain over time a whitelist of tests, which complicates the (already convoluted) CI setup. I'm also worried we might miss some side-effects of having both frameworks installed.

What time do those tests take?

**Comment by sshleifer** [comment]
6 mins / 13 mins, at least once per PR.
I think it's a good tradeoff.
- we are literally losing 0 coverage.
- people rarely add new tests that check equivalence of torch and tf.  We would rarely need to update the whitelist.
- For most bugs, we would all be able to reproduce it locally where we have both frameworks installed.
Don't feel too strongly, of course.
- Also happy to move this

**Comment by julien-c** [comment]
What does 6 mins / 13 mins mean?

**Comment by sshleifer** [comment]
Before this change `run_test_torch_and_tf` takes 6 mins, and the whole CircleCI suite takes 13mins.

**Comment by julien-c** [comment]
How do you measure the whole CircleCI suite, given that the tests are (AFAIK) parallelized?

**Comment by sshleifer** [comment]
By looking at circleci output like [this](https://app.circleci.com/pipelines/github/huggingface/transformers/5174/workflows/3aecb0e9-14f0-4762-ad6a-73b60db2d626) and guessing

**Comment by sshleifer** [comment]
[This] (https://app.circleci.com/pipelines/github/huggingface/transformers/6581/workflows/b114b306-2553-42cf-bb67-ba507e051929) is after pipelines improvements. I may be overestimating the benefit of the change.

**Comment by sshleifer** [comment]
@julien-c seems like we don't want to do this? Happy to abandon.

File: .circleci/config.yml (hunk 1 above)
Comment (julien-c) [reply .circleci/config.yml:16]: I find this whitelist tricky to maintain into the future. Maybe we can simplify it with multiple `-k` filters (in a single command)

Comment (sshleifer) [reply .circleci/config.yml:16]: pytest doesn't support multiple -k filters yet https://stackoverflow.com/questions/13295627/can-py-test-support-multiple-k-options

**Pull request closed without merging**